block/manager.go

```diff
@@ -225,3 +228,12 @@ func (m *Manager) syncBlockManager() error {
 	m.logger.Info("Synced.", "current height", m.State.Height(), "last submitted height", m.LastSubmittedHeight.Load())
 	return nil
 }
+
+func (m *Manager) updateTargetHeight(h uint64) {
+	for {
+		currentHeight := m.TargetHeight.Load()
+		if m.TargetHeight.CompareAndSwap(currentHeight, max(currentHeight, h)) {
```

Comment:
if `max(currentHeight, h)` is `currentHeight`
u'll call `m.TargetHeight.CompareAndSwap(currentHeight,currentHeight)`
won't it return `false` and be stuck forever?